### PR TITLE
Event/2023pncmp

### PR DESCRIPTION
**Pull request by rutmanz**

### CODE REVIEW

File: src/main/java/org/team1540/robot2023/commands/auto/AutoCone.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                                        new GrabberAggressiveCommand(intake)
+                                ),
+                                new WaitUntilCommand(() -> controller.getLeftTriggerAxis() > 0.95).unless(() -> controller == null),
+                                new SetArmPosition(arm, AutoHybrid.catchNull(positions.score)).unless(() -> positions.score == null),
+                                Commands.deadline(
+                                        Commands.sequence(
+                                                new WaitCommand(0.25),
+                                                new SetArmPosition(arm, AutoHybrid.catchNull(positions.retreat)).unless(() -> positions.retreat == null),
+                                                new ResetArmPositionCommand(arm, false),
+                                                new ResetArmPositionCommand(arm, true)
```
Comment (mimizh2418) [inline src/main/java/org/team1540/robot2023/commands/auto/AutoCone.java:41]: Why is this happening once without reset and once with reset?

File: src/main/java/org/team1540/robot2023/commands/auto/AutoCube.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                                                new GrabberOuttakeCommand(intake)
+
+                                        )
+                                ),
+                                //                                    new WaitUntilCommand(() -> controller.getLeftTriggerAxis() > 0.95).unless(() -> controller == null || positions.polePosition == PolePosition.CENTER),
+                                Commands.deadline(
+                                        Commands.sequence(
+                                                new SetArmPosition(arm, AutoHybrid.catchNull(positions.retreat)).unless(() -> positions.retreat == null),
+                                                new ResetArmPositionCommand(arm, false),
+                                                new ResetArmPositionCommand(arm, true)
```
Comment (mimizh2418) [inline src/main/java/org/team1540/robot2023/commands/auto/AutoCube.java:40]: Same as above, probably not necessary to set the position without zero and set it again with zero

File: src/main/java/org/team1540/robot2023/commands/auto/sequence/Auto2PieceTaxiConeVision.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.team1540.robot2023.utils.Limelight;
+import org.team1540.robot2023.utils.PolePosition;
+import org.team1540.robot2023.utils.ScoringGridLocation;
+
+import java.util.List;
+
+public class Auto2PieceTaxiConeVision extends AutoCommand {
+    public Auto2PieceTaxiConeVision(Drivetrain drivetrain, Arm arm, WheeledGrabber intake, ScoringGridLocation.OuterGrid grid, Limelight limelight) {
+        List<Command> pathCommands = getPathPlannerDriveCommandGroup(drivetrain, grid.getPathName("2PieceTaxiConeVision"), new PathConstraints(4, 2), true);
+        limelight.setPipeline(Limelight.Pipeline.GAME_PIECE);
```
Comment (mimizh2418) [inline src/main/java/org/team1540/robot2023/commands/auto/sequence/Auto2PieceTaxiConeVision.java:28]: Pipeline switching should be in `addCommands`

File: src/main/java/org/team1540/robot2023/commands/auto/sequence/AutoBottomGrid2PieceTaxiVision.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.util.List;
+
+public class AutoBottomGrid2PieceTaxiVision extends AutoCommand {
+    public AutoBottomGrid2PieceTaxiVision(Drivetrain drivetrain, Arm arm, WheeledGrabber intake, Limelight limelight) {
+        List<Command> pathCommands = getPathPlannerDriveCommandGroup(drivetrain,"BottomGrid2PieceTaxiVision", new PathConstraints[]{
+                new PathConstraints(2,1),
+                new PathConstraints(4,2)
+        }, false);
+        limelight.setPipeline(Limelight.Pipeline.GAME_PIECE);
```
Comment (mimizh2418) [inline src/main/java/org/team1540/robot2023/commands/auto/sequence/AutoBottomGrid2PieceTaxiVision.java]: Goes in `addCommands`

File: src/main/java/org/team1540/robot2023/commands/auto/sequence/AutoBottomGrid2_5PieceTaxiConeVision.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.util.List;
+
+public class AutoBottomGrid2_5PieceTaxiConeVision extends AutoCommand {
+    public AutoBottomGrid2_5PieceTaxiConeVision(Drivetrain drivetrain, Arm arm, WheeledGrabber intake, Limelight limelight, Limelight frontLimelight) {
+        List<Command> pathCommands = getPathPlannerDriveCommandGroup(drivetrain,"BottomGrid2_5PieceTaxiConeVision", new PathConstraints[]{
+                new PathConstraints(2,1),
+                new PathConstraints(4,2)
+        }, true);
+        limelight.setPipeline(Limelight.Pipeline.GAME_PIECE);
```
Comment (mimizh2418) [inline src/main/java/org/team1540/robot2023/commands/auto/sequence/AutoBottomGrid2_5PieceTaxiConeVision.java]: goes in `addCommands`

File: src/main/java/org/team1540/robot2023/commands/auto/sequence/AutoTopGrid2PieceVision.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import org.team1540.robot2023.utils.AutoCommand;
+import org.team1540.robot2023.utils.Limelight;
+import org.team1540.robot2023.utils.PolePosition;
+
+import java.util.List;
+
+public class AutoTopGrid2PieceVision extends AutoCommand {
+    public AutoTopGrid2PieceVision(Drivetrain drivetrain, Arm arm, WheeledGrabber intake, Limelight limelight) {
+        List<Command> pathCommands = getPathPlannerDriveCommandGroup(drivetrain, "TopGrid2PieceVision");
+        limelight.setPipeline(Limelight.Pipeline.GAME_PIECE);
```
Comment (mimizh2418) [inline src/main/java/org/team1540/robot2023/commands/auto/sequence/AutoTopGrid2PieceVision.java]: goes in `addCommands`

**Review by mimizh2418 — COMMENTED** [review]
There's some weird stuff going on with the arm position resetting that might not be necessary, and also pipeline switching is implemented incorrectly in some of the auto routines that have vision. It might be helpful to fully delete the ones we don't plan on using/the ones that don't work.

**Comment by rutmanz** [comment]
I mean those are valid concerns, but we don't run those autos ever, and for the reset arm part, I don't want to break something that's working without testing it first, and this isn't a great time to be doing that.

**Pull request merged**